geth/transactions/queue/queue.go

```diff
@@ -133,9 +133,6 @@ func (q *TxQueue) Reset() {
 // Enqueue enqueues incoming transaction
 func (q *TxQueue) Enqueue(tx *common.QueuedTx) error {
 	log.Info(fmt.Sprintf("enqueue transaction: %s", tx.ID))
-	if (tx.Hash != gethcommon.Hash{} || tx.Err != nil) {
```

Comment:
Ok, I remember when it can happen that a transaction is queued multiple times. It's possible from the interface. We should check if the given transaction is not in the `q.transactions` already.

Reply:
you meant transaction managers interface, right? not one that is used by react,  i don't see how react can set ID for queued transaction